vpr/src/pack/pack.cpp

```diff
@@ -167,9 +178,28 @@ bool try_pack(t_packer_opts* packer_opts,
             VTR_LOG("Packing failed to fit on device. Re-packing with: unrelated_logic_clustering=%s balance_block_type_util=%s\n",
                     (allow_unrelated_clustering ? "true" : "false"),
                     (balance_block_type_util ? "true" : "false"));
+        } else if (pack_iteration == 1 && floorplan_not_fitting) {
+            VTR_LOG("Floorplan regions are overfull: trying to pack again using cluster attraction groups. \n");
+            attraction_groups.create_att_groups_for_overfull_regions();
+            attraction_groups.set_att_group_pulls(1);
+
+        } else if (pack_iteration >= 2 && pack_iteration < 5 && floorplan_not_fitting) {
+            VTR_LOG("Floorplan regions are overfull: trying to pack again with more attraction groups exploration and higher target pin utilization. \n");
+            attraction_groups.create_att_groups_for_overfull_regions();
+            VTR_LOG("Pack iteration is %d\n", pack_iteration);
+            attraction_groups.set_att_group_pulls(4);
```

Comment:
Think about using defined constants for all these numbers (2, 5, 4 ...).  Can define in a block right above this code and comment what each does and why they have the values they do.